include/aikido/constraint/uniform/detail/RnBoxConstraint-impl.hpp

```diff
@@ -175,19 +175,40 @@ std::vector<ConstraintType> RBoxConstraint<N>::getConstraintTypes() const
 //==============================================================================
 template <int N>
 bool RBoxConstraint<N>::isSatisfied(
-    const statespace::StateSpace::State* state) const
+    const statespace::StateSpace::State* state, TestableOutcome* outcome) const
 {
+  DefaultOutcome* defaultOutcomeObject = nullptr;
+  if (outcome)
+  {
+    defaultOutcomeObject = dynamic_cast<DefaultOutcome*>(outcome);
+    if (!defaultOutcomeObject)
+      throw std::invalid_argument(
+          "TestableOutcome pointer is not of type DefaultOutcome.");
+  }
```

Comment:
Nit: This logic is duplicated in quite a few places. I suggest putting it in a helper function, e.g.:
```c++
template <class Child, class Parent>
Child* checked_dynamic_cast<Child, Parent>(Parent* _parent)
{
  auto child = dynamic_cast<Child*>(_parent);
  if (!child && _parent)
  {
    std::stringstream message;
    message << "'" << std::type_info<Parent>::name() << "' is not of type "'"
            << std::type_info<Child>::name() <<"'.";
    throw std::invalid_argument{
  }
}
```
Then, you could reduce all of this boilerplate to just:
```c++
auto defaultOutcomeObject = checked_dynamic_cast<DefaultOutcome>(outcome);
```
(Some massaging may be required for the template type inference to work.)

Note that if you make `clear()` part of the base `TestableOutcome` API, then you could also put `clear()` inside that helper function! 😍